### ISSUE TITLE
Extra character inserted when writing ~  or ` with vscode and french keyboard layout

### ISSUE THREAD

**Issue by vlj** [microsoft/terminal#391]
* Your Windows build number: My build is : Microsoft Windows [version 10.0.18362.1]
Using vscode 1.32.3 with conpty enabled.

* What you're doing and what's happening: 
When trying to type some specific character with a french keyboard layout, an extra character ("é") is added. For now I spotted 2 characters, ~ and `.
More precisely, if I press "alt gr" and 2 at the same time, a "é" is added. If I then press space, then I have "é~" in the prompt.
If I press "alt gr" and seven at the same time, a "é" is added. If I then press space, then I have "é`" in the prompt.

![3Pqo1WSEh2](https://user-images.githubusercontent.com/6413475/54755361-18137380-4be6-11e9-9db4-587e530826d5.gif)

* What's wrong / what should be happening instead:
What should happen is that no ~ is added.

**Comment by vlj** [comment]
Note that it doesn't happen if I disable conpty.
I didn't find a similar issue (it's not a newline)

**Comment by zadjii-msft** [comment]
I've filed MSFT:20921862 internally to make sure this gets investigated.

**Comment by vlj** [comment]
I don't know where the issue is in the stack, but just in case, it's still present with Terminal app (master, checked yesterday).

**Comment by LiHRaM** [comment]
Using Danish keyboard results in invisible character being inserted when using alt gr to compose letters.

```
# cmd.exe
C:\Users\hilma\git>ver
Microsoft Windows [Version 10.0.18922.1000]
C:\Users\hilma\git>¨~ 
```

```
# wsl
$ cd ~
-bash: cd: $'\250~': No such file or directory
```

In practice I've had to switch to other terminals to use | (pipes) and vim, as it exists insert mode whenever I compose with alt gr.

**Comment by slayerizer** [comment]
I cannot use any characters composed with ALT GR..        including @ and [ ]

**Comment by fle108** [comment]
still present vscode 1.38.1

**Comment by DHowett-MSFT** [comment]
Yes, this was fixed by #4192. Verified in VSCode, Putty, and Terminal. VSCode now produces `qq` for <kbd>AltGr-2 Q</kbd>, but no other terminal emulators using the same infrastructure do so. Any further bugs here are liable to be upstream's fault.

**Comment by DHowett-MSFT** [comment]
FWIW: I can't even reproduce this with the original inbox console host. Perhaps this was always VS Code's fault.
